appcues/src/main/java/com/appcues/ui/composables/AppcuesComposition.kt

```diff
@@ -133,3 +133,12 @@ private fun BoxScope.ComposeLastRenderingState(state: Rendering) {
         }
     }
 }
+
+@Composable
+private fun BoxScope.ApplyBackgroundDecoratingTraits(list: List<BackdropDecoratingTrait>) {
+    // get last trait if its not null compose it and drop last calling it again recursively
+    val item = list.lastOrNull()
+    if (item != null) {
+        with(item) { BackdropDecorate { ApplyBackgroundDecoratingTraits(list.dropLast(1)) } }
+    }
```

Comment:
oh clever ok, so going in reverse like this makes sure that the first trait is actually at the deepest level inside this composition tree right?

Reply:
yes this basically means that if:

traits {
  backdrop
  keyhole
}

keyhole is the parent of the composition here. so we can draw the backdrop INSIDE the keyhole trait, giving us the ability to crop everything in there.